README.md

```diff
@@ -16,3 +16,14 @@ To run tests:
 ```sh
 swift test
 ```
+
+### Developer's Guide
```

Comment:
"Developer's Guide" is ambiguous, since people using the library are developers too

Maybe:
- Contributor's Guide
- Internal Developer's Guide
- Guide for Developers of aws-crt-swift
- ???